source/sggsym.h

```diff
@@ -0,0 +1,116 @@
+#ifndef SGGSYM_H
+#define SGGSYM_H
+
+#include "Symbiont.h"
+#include "SymWorld.h"
+
+class SggSymbiont: public Symbiont {
+protected:
+  double sgg_donate = 0;
+
+
+public:
+  SggSymbiont(emp::Ptr<emp::Random> _random, emp::Ptr<SymWorld> _world, emp::Ptr<SymConfigBase> _config, double _intval=0.0, double _donation = 0.0, double _points = 0.0 ) : Symbiont(_random, _world, _config, _intval, _points),sgg_donate(_donation)
+  {}
+  SggSymbiont(const SggSymbiont &) = default;
+  SggSymbiont(SggSymbiont &&) = default;
+  SggSymbiont() = default;
+
+
+  SggSymbiont & operator=(const SggSymbiont &) = default;
+  SggSymbiont & operator=(SggSymbiont &&) = default;
+
+
+  double Getdonation() {return sgg_donate;}
+  void Setdonation(double _in) {sgg_donate = _in;}
+
+  void mutate(){
+    // double pre_value = interaction_val;
+    if (random->GetDouble(0.0, 1.0) <= mut_rate) {
+      interaction_val += random->GetRandNormal(0.0, mut_size);
+      if(interaction_val < -1) interaction_val = -1;
+      else if (interaction_val > 1) interaction_val = 1;
+    }
+    if (random->GetDouble(0.0, 1.0) <= mut_rate) {
+      double mutate_size = random->GetRandNormal(0.0, mut_size);
```

Comment:
You can go back to doing this all on one line (`sgg_donate += random->GetRandNormal(0.0, mut_size);`. We just had it this way for debugging